src/vm.rs

```diff
@@ -214,3 +224,12 @@ pub trait Precompile: std::fmt::Debug {
         Option<(Vec<MemoryQuery>, Vec<MemoryQuery>, Vec<Self::CycleWitness>)>,
     );
 }
+
+pub enum SpongeExecutionMarker {
+    MemoryQuery,
+    DecommittmentQuery,
+    StorageLogReadOnly,
+    StorageLogWrite,
+    CallstackPush,
+    CallstackPop,
+}
```

Comment:
all looks good, but did this maybe pop up since you based off a different branch?

Reply:
Yep, rebased to `dl-precompiles`